[incubator/zookeeper] Support Zookeeper official (3.5.4) image

donbowman opened the pull request:
The existing chart is using gcr.io/google_samples/k8szk, which is not
maintained.

The Zookeeper project has an 'official' image on DockerHub, based
on Zookeeper 3.5.4(-beta). The format of the environment variables
and scripts is different.

Existing issuse such as [#3020](https://github.com/helm/charts/issues/3020)
went stale.

Add support for the Zookeeper official image, mapping the environment
variables and paths.

Signed-off-by: Don Bowman <db@donbowman.ca>

<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x ] Chart Version bumped
- [x ] Variables are documented in the README.md

donbowman committed:
[incubator/zookeeper] Support Zookeeper official (3.5.4) image

The existing chart is using gcr.io/google_samples/k8szk, which is not
maintained.

The Zookeeper project has an 'official' image on DockerHub, based
on Zookeeper 3.5.4(-beta). The format of the environment variables
and scripts is different.

Existing issuse such as [#3020](https://github.com/helm/charts/issues/3020)
went stale.

Add support for the Zookeeper official image, mapping the environment
variables and paths.

Signed-off-by: Don Bowman <db@donbowman.ca>

fabriziofortino reviewed (commented):
That's great work. Should this point to the latest stable release (3.4.13) since 3.5 is still in beta?

donbowman committed:
[incubator/zookeeper] Change default zookeeper image to 3.4.13

3.5 is not yet released, 3.4.13 is the latest stable,
change the default.

Signed-off-by: Don Bowman <db@donbowman.ca>

davidkarlsen commented on incubator/zookeeper/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
-version: 1.3.0
-appVersion: 3.4.10
+version: 2.0.0
+appVersion: 3.5.4-beta
```
```suggestion
appVersion: 3.4.13
```

davidkarlsen replied:
And version 3.4.14 seems to be out: https://zookeeper.apache.org/releases.html

davidkarlsen commented on incubator/zookeeper/README.md:
```diff
@@ -6,7 +6,7 @@ This helm chart provides an implementation of the ZooKeeper [StatefulSet](http:/
 * PersistentVolume support on the underlying infrastructure
 * A dynamic provisioner for the PersistentVolumes
-* A familiarity with [Apache ZooKeeper 3.4.x](https://zookeeper.apache.org/doc/current/)
+* A familiarity with [Apache ZooKeeper 3.5.x](https://zookeeper.apache.org/doc/r3.5.4-beta/)
```
Need consistent version here too

donbowman replied:
ok i'll make these changes and rebase

donbowman committed:
fix: bump to 3.5.5 by default, update README.md to reflect

Signed-off-by: Don Bowman <db@donbowman.ca>

kppullin commented on incubator/zookeeper/values.yaml:
```diff
@@ -137,7 +112,7 @@ command:
 
 persistence:
-  enabled: true
+  enabled: false
```
I was curious as to what would happen when upgrading a deployment of this chart that was originally configured with persistence enabled, with my main concern being that the existing persistent volumes would be deleted.

Fortunately, k8s rejected the change with this error `Error: UPGRADE FAILED: StatefulSet.apps "zookeeper" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden`, preventing any data loss.

Generally, what's the motivation for changing this from `true` to `false`?

donbowman replied:
reverted. IMHO the default is wrong but it was not the intent of this PR to change.

donbowman committed:
fix: revert persistence storage default to true

The intent of PR11343 was just to enale to work with
upstream container, not change this default. IMHO
the default is incorrect (true without setting the
storage class or size usually gives incorrect results).

Signed-off-by: Don Bowman <db@donbowman.ca>

donbowman committed:
Merge branch 'master' into zookeeper-to-default-image
